components/BidderBio.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from 'react';
+import Text, { textStyle } from './Text';
+import { Wallet } from '../server/api/types';
+import { useAccount, useSignMessage } from 'wagmi';
+import { verifyMessage } from '@wagmi/core';
+import { trpc } from '../utils/trpc';
+import { config, formatAddress } from '../utils/utils';
+
+export default function BidderBio({ bidder }: { bidder: Wallet }) {
+  const { address } = useAccount();
+  const { signMessageAsync } = useSignMessage();
+
+  const mutation = trpc.private.insertBio.useMutation();
+  const author = trpc.walletData.useQuery({
+    address: address?.toLowerCase() || '',
+  });
+
+  const [bio, setBio] = useState(bidder.bioText || '');
+  const [message, setMessage] = useState<{ text: string; isSuccess: boolean } | null>(null);
+
+  let isAuthor = false;
+
+  if (bidder.address === address?.toLowerCase()) {
+    isAuthor = true;
+  } else {
+    isAuthor = author.data?.details?.isAuthor || false;
+  }
+
+  const signAction = async () => {
+    try {
+      const signature = await signMessageAsync({ message: bio });
+      if (signature) {
+        const isValid = await verifyMessage(config, {
+          address: address || '0x',
+          message: bio,
+          signature,
+        });
+        if (isValid) {
+          await mutation.mutateAsync({
+            wallet: {
+              address: author.data?.details.address || '',
+              isAuthor,
+            },
+            bidderAddress: bidder.address.toString().toLowerCase(),
+            bioText: bio,
+          });
+          setMessage({ text: 'Bio added.', isSuccess: true });
+        } else {
+          setMessage({ text: 'Validation failed.', isSuccess: false });
+        }
+      }
```

Comment:
You are doing signature verification on the web app. This is not adding any security, since anyone can just call this mutation with `{ isAuthor: true }` and do whatever they want.

Instead, get the signature and send it to the server side. On the server, call `verifyMessage` to make sure the signature is correct and is coming from the right address